src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -1344,4 +1345,83 @@ public void setConqueredDays(int conqueredDays) {
 	public int getConqueredDays() {
 		return this.conqueredDays;
 	}
+	
+	public List<TownBlock> getTownBlocksForPlotGroup(PlotGroup group) {
+		
+		ArrayList<TownBlock> retVal = new ArrayList<>();
+		
+		TownyMessaging.sendErrorMsg(group.toString());
+		
+		for (TownBlock townBlock : getTownBlocks()) {
+			if (townBlock.hasPlotGroup() && townBlock.getPlotGroup().equals(group))
+				retVal.add(townBlock);
+		}
+		
+		return retVal;
+	}
+	
+	public void addPlotGroup(PlotGroup group) {
+		if (!hasGroups()) 
+			this.plotGroups = new HashSet<>();
+		
+		TownyMessaging.sendErrorMsg("Adding " + group.getGroupName() + " to " + this.getName());
+		
+		this.plotGroups.add(group);
+		
+	}
+	
+	public void removePlotGroup(PlotGroup plotGroup) {
+		if (hasGroups() && hasGroup(plotGroup)) {
+			for (TownBlock tb : getTownBlocks()) {
+				if (tb.hasPlotGroup() && tb.getPlotGroup().equals(plotGroup)) {
+					tb.getPlotGroup().setID(null);
+					TownyUniverse.getInstance().getDataSource().saveTownBlock(tb);
+				}
+			}
+		}
+	}
+	
+	public void setPlotGroups(String str) {
+		
+		if (plotGroups == null)
+			plotGroups = new HashSet<>();
+		
+		String[] groups = str.split(";");
+		
+		for (String groupStr : groups) {
+			addPlotGroup(PlotGroup.fromString(groupStr));
+		}
+	}
+	
+	public int generatePlotGroupID() {
+		return (hasGroups()) ? getGroups().size() : 0;
+	}
+
+	@Override
+	public HashSet<PlotGroup> getGroups() {
+		return plotGroups;
+	}
+
+	@Override
+	public PlotGroup getGroupFromID(int ID) {
+		if (hasGroups()) {
+			for (PlotGroup pg : getGroups()) {
+				if (pg.getID() == ID) 
+					return pg;
+			}
+		}
+		
+		return null;
+	}
```

Comment:
Both methods should be removed to getPlotsGroups and getPlotGroupFromID to prevent confusion for other developers who might think Towny has permission groups.